lib/clickhousex/query.ex

```diff
@@ -89,10 +89,12 @@ defimpl DBConnection.Query, for: Clickhousex.Query do
 
   defp query_type(statement) do
     cond do
+      # `INSERT` comes first for `INSERT INTO ... SELECT FROM ...` to be marked as insert
       String.starts_with?(statement, "INSERT") -> :insert
+      # `select` comes before `insert` for `SELECT ... FROM ... WHERE query_kind = 'Insert'` to be marked as select
+      Regex.match?(@select_query_regex, statement) -> :select
       Regex.match?(@insert_query_regex, statement) -> :insert
       Regex.match?(@create_query_regex, statement) -> :create
-      Regex.match?(@select_query_regex, statement) -> :select
```

Comment:
This was the reason for https://github.com/plausible/analytics/actions/runs/3204425998/jobs/5235698705 fail. The [query in the failing test](https://github.com/plausible/analytics/blob/aadf892a71f6db4f7f1e0d961bfc4d7296fa2b59/test/plausible/google/vcr_test.exs#L9-L17) had "Insert" in the `where` clause which made it be classified under ":insert" which resulted in skipped [`FORMAT JSONCompact`](https://github.com/plausible/clickhousex/blob/f0e0820fe79f7e4fa433497bf41fa2e910e51eb0/lib/clickhousex/http_client.ex#L157-L159) and clickhouse responding in tsv (`"3077\n\n"`) and [the match fail.](https://github.com/plausible/clickhousex/blob/f0e0820fe79f7e4fa433497bf41fa2e910e51eb0/lib/clickhousex/codec/json.ex#L42)

Reply:
This is now more similar [to current `fork/master`:](https://github.com/plausible/clickhousex/blob/e6b7cc5a1ea7bb26d77af8fdae03952f662469af/lib/clickhousex/query.ex#L87-L95)

```elixir
  defp query_type(statement) do
    cond do
      String.starts_with?(statement, "INSERT") -> :insert
      Regex.match?(@select_query_regex, statement) -> :select
      Regex.match?(@insert_query_regex, statement) -> :insert
      Regex.match?(@alter_query_regex, statement) -> :alter
      true -> :update
    end
  end
```

Reply:
(I don't know why `:created` was dropped)